packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5553,6 +5553,62 @@ class ExcludeSemantics extends SingleChildRenderObjectWidget {
   }
 }
 
+/// A widget that annotates the child semantics with an index.
+///
+/// Certain widgets like the [ListView] will automatically provide a child
+/// index for building semantics. A user may wish to manually provide semanitc
+/// indexes if not all child of the scrollable contribute semantics.
+///
```

Comment:
Maybe add: Semantic indexes are used by TalkBack/Voiceover to make announcements about the current scroll state, e.g. "<example announcement>".